fix: try refreshing cookie on multiple cases

StephaneRob opened the pull request:
Currently we try to refresh cookie only when the access token is found and is expired.
If the refresh cookie is present we should try refreshing session with it even if invalid or not found.

yordis commented:
Would you mind adding an entry in the changelog https://github.com/ueberauth/guardian/blob/master/CHANGELOG.md

yordis reviewed (changes requested):
Please add some entry in the changelog so we keep the PR ready to be release

StephaneRob commented:
Hi @yordis, I added some test cases and updated the CHANGELOG.
The changelog doesn't mention the neither `refresh_from_cookie` option nor `VerifyCookie` plug deprecation. Do you want I add it under `2.2.0` version?

yordis commented:
@StephaneRob please if you don't mind I would appreciate updating the CHANGELOG for sure

StephaneRob commented:
@yordis done.

yordis commented:
You are the best! Thank you so much!
